src/Container/GenericContainer.php

```diff
@@ -54,17 +57,15 @@ public function config(object $config): self
 
     public function get(string $className): object
     {
-        try {
-            return $this->resolve($className);
-        } catch (Throwable $throwable) {
-            // TODO: Logging
+        $this->log->startResolving();
```

Comment:
Probably not something to fix before merging this PR, but something to keep in mind...

If I have `ClassA` with a dependency on `ClassB` that has a dependency on `ClassC` and if `ClassC` looks like the following:

```php
class ClassC
{
    private SomeDependency $blah;

    public function __construct(Container $container)
    {
        $this->blah = $container->get(SomeDependency::class);
    }
}
```

Won't that reset my log for the autowiring of `ClassA`? I think the fix for this would be to store the originator information in a context as well, so that it is not tied to the master log class.

There's a chance this is what was affecting initializers as well?

Reply:
That's exactly the problem initializers also have, yeah. You'd need to be able to distinguish whether a `container->get` call is made by the user, or originates from a nested container call.

I actually solved this problem in untitled by implementing two containers. They were the same (with a trait), except that one implementation didn't reset the log on `get`. I called this the `ContainerForInitializers` implementation, and they shared the same log instance.

Whenever I passed the container to an initializer from within the container, I'd give it that other implementation that didn't reset the log on `get`, we could do the same for injected container instances to solve this problem.

However, maybe it's easier to check `debug_backtrace` and determine whether we reset the log or not based on that info?

> I think the fix for this would be to store the originator information in a context as well, so that it is not tied to the master log class.

Not sure I understand, maybe this is an even better solution.

It would be nice if we could fix this, but let's first merge this PR 👍

Reply:
I made an issue for it: https://github.com/tempestphp/tempest-framework/issues/180